mako-templates/committing.html

```diff
@@ -0,0 +1,174 @@
+
+<%self:code lang="markdown" write="README.md">
+# reponame
+This is a better description of this repository. Imagine you just wrote it
+into your text editor.
+
+More text here. Lorem ipsum blah blah blah.
+</%self:code>
+
+<p>Remember to save it in the editor. Then run <code>git status</code> again:
+
+<%self:runcommands>
+$ git status
+</%self:runcommands>
+
+<p>Adding, also known as staging, means choosing what will be included in the next commit.
+This way it's possible to edit several files and commit only some of the changes.
+Adding a file doesn't modify it;
+it just makes Git remember the changes you did.
+
+<p>As the status message suggests, we will use <code>git add <file>...</code>.
+Here <code><file></code> means that you should put a file name there,
+and <code>...</code> means that you can specify several files if you want.
+We will add only one file, the README.md:
+
+<%self:runcommands>
+$ git add README.md
+$ git status
+$ git diff --staged
+</%self:runcommands>
+
+<p>The first line of the README is <code># reponame</code>, and it was not changed.
+There was one line after that, <code>The description ...</code>, and it was deleted.
+The rest was added when editing the file in the editor.
+
+<p>Diffing is optional, and it does not affect the output of <code>git status</code>,
+but it may help you discover problems.
+If you notice that something isn't quite right, you can still edit the files,
+and then run <code>git add</code> again when you are done.
+Alternatively, if you don't want to commit any changes to <code>README.md</code>,
+you can use <code>git restore</code> as shown in <code>git status</code> output to undo the <code>git add</code>:
+
+<%self:runcommands>
+$ git status
+$ git restore --staged README.md
+$ git status
+$ git add README.md
+$ git status
+$ git commit -m "add better description to README"
+</%self:runcommands>
+
+<p>This error happens when you try to commit for the first time.
+To fix this, run the commands that the error message suggests,
+replacing <code>Your Name</code> with your GitHub username
+and <code>you@example.com</code> with the email address you used when creating your GitHub account.
+
+<%self:runcommands>
+$ git config --global user.email "you@example.com"
+$ git config --global user.name "yourusername"
+</%self:runcommands>
+
+<p>Now committing should work:
+
+<%self:runcommands>
+$ git commit -m "add better description to README"
+</%self:runcommands>
+
+<p>The text after <code>-m</code> is a <strong>commit message</strong>.
+Enter something descriptive, such as "add better description to README" in this case;
+finding something from a long list of commits really sucks if the message of every commit is "files edited".
+
+<p>Now the file no longer shows up as modified in <code>git status</code>, because we committed the change:
+
+<%self:runcommands>
+$ git status
+</%self:runcommands>
+
+<p>You can ignore the <code>main</code> part of <code>origin/main</code> for now,
```

Comment:
I don't like `<code>` compared to markdown-style backticks. Is there some way to use backticks?